### ISSUE TITLE
custom_columns feature request

### ISSUE THREAD

**Issue by joshrabinowitz** [bigpresh/Dancer-Plugin-SimpleCRUD#62]
Sometimes when I define a custom_column I'd like to combine data from multiple columns, but the custom_columns spec only accepts one input column name. 

I'd love to be able to show a custom column that aggregates data from multiple columns.  Perhaps something like:

    custom_columns => {
        result   => { raw_columns=>[qw(id,name,other)],          transform=>sub { ... }},
         # or overload meaning of 'raw_column' if it's an arrayref
    },

If there's no pushback I can work up a patch @bigpresh

**Comment by msouth** [comment]
This isn't an objection, just a comment--if you pass the id in, your "transform" can retrieve the row and do whatever it wants with however many columns it wants.

One option might be to just point that fact out in the documentation; with maybe a "full_name" column that takes the id as the raw column, retrieves the row and returns firstname space lastname.

**Comment by joshrabinowitz** [comment]
hm, yeah, that's a possibility.

I'm trying to minimize round-trips for data that we've already fetched. Otherwise showing a 500-row table now requires 501 sql round trips instead of 1.

**Comment by bigpresh** [comment]
I think this is already doable, albeit not nicely documented.  The `custom_columns` option sets up per-cell callbacks in HTML::Table::FromDatabase using its `callbacks` option; since version 0.04, each callback receives the value of the named column (which we take from the `raw_column` attribute), but also the full row hashref, so you could get other values, too.

So I think you could say e.g.:
```perl
...
custom_columns => {
    saluation => {
        raw_column => 'name',
        transform => sub {
            my ($name_value, $row) = @_;
            return "Hi, $row->{title} $name_value!",
        },
},
```

I've not tested that, but if we document how the `custom_columns` option is implemented a bit better, mentioning that under the hood it just sets up HTML::Table::FromDatabase callbacks, it would likely be a bit clearer that such tomfoolery is possible :)

**Comment by joshrabinowitz** [comment]
I just tested using 

````
custom_columns => {
    dumped => {
        raw_column => 'id',
        transform => sub { my ($s, $row, @more) = @_; dump($row); },
    }
}
````

and it looks good. 

It's such a useful feature that I agree that it would be good to document. See https://github.com/bigpresh/Dancer-Plugin-SimpleCRUD/pull/66

**Comment by joshrabinowitz** [comment]
@bigpresh I consider this issue (and #65 ) closed now that 1.00 is released.  Agree?

**Comment by bigpresh** [comment]
Agreed - closing!
